src/Aspire.Hosting/ApplicationModel/UserSecretsParameterDefault.cs

```diff
@@ -24,7 +25,7 @@ public override string GetDefaultValue()
         var configurationKey = $"{ParameterResourceBuilderExtensions.ConfigurationSectionKey}:{parameterName}";
         if (!TrySetUserSecret(applicationName, configurationKey, value))
         {
-            throw new DistributedApplicationException($"Failed to set value for parameter '{parameterName}' in application '{applicationName}' to user secrets.");
+            Debug.WriteLine($"Failed to set value for parameter '{parameterName}' in application '{applicationName}' to user secrets.");
```

Comment:
Why?

Reply:
To make it a bit easier to see if it's doesn't succeed during dev. There's no logging at this stage and swallowing blindly just seems bad mmm/'k

Reply:
Would it be possible for us to get the ability to log here? (obviously not in this PR) It seems like a big hole that we aren't able to log things until the app is built. Especially as we add more functionality earlier in the app. Our only options are throw or eat...

Reply:
Another cause for this to not succeed is if `appHostAssembly is null`. Do we need to support that scenario? It feels odd that we would Debug.WriteLine if someone passed null (which we claim to support). We basically wouldn't do anything.

Reply:
Logged #4597 to track enabling logging pre-host build.

Reply:
> Another cause for this to not succeed is if appHostAssembly is null. Do we need to support that scenario? 

@eerhardt I made it non-nullable and don't call in now if the appHostAssembly is null for some reason.

Reply:
Now that it's internal, it is less of a concern. But this sounds good.